contracts/vesting/VestingEscrowLL.vy

```diff
@@ -212,3 +228,62 @@ def collect_dust(token: ERC20, beneficiary: address = msg.sender):
         amount = amount + self.total_claimed - self._total_vested_at(self.disabled_at)
 
     assert token.transfer(beneficiary, amount, default_return_value=True)
+
+
+# Liquid locker specific functions
+
+@external
+def set_signed_message(_hash: bytes32, _signed: bool):
+    """
+    @notice Mark a message as signed
+    @param _hash Message hash
+    @param _signed True: signed, False; not signed
+    @dev Can only be called by operators
+    """
+    assert msg.sender == self.recipient
+    assert _hash != empty(bytes32)
+    self.messages[_hash] = _signed
+    log SetSignedMessage(_hash, _signed)
+
+
+@external
+def set_operator(_operator: address, _flag: bool):
+    """
+    @notice Add or remove an operator
+    @param _operator Operator
+    @param _flag True: operator, False: not operator
+    @dev Can only be called by recipient
+    @dev Requires prior approval by factory owner
+    """
+    assert msg.sender == self.recipient
+    assert _operator != empty(address)
+    if _flag:
+        assert self.factory.operators(self.token.address, _operator)
+    self.operators[_operator] = _flag
+    log SetOperator(_operator, _flag)
+
+
+@external
+@view
+def isValidSignature(_hash: bytes32, _signature: Bytes[128]) -> bytes4:
+    """
+    @notice Check whether a message should be considered as signed
+    @param _hash Hash of message
+    @param _signature Signature, unused
+    @return EIP-1271 magic value
+    """
+    assert self.messages[_hash] and len(_signature) == 0
+    return EIP1271_MAGIC_VALUE
+
+
+@external
+@payable
+def call(_target: address, _data: Bytes[2048]):
+    """
+    @notice Call another contract through the escrow contract
+    @param _target Contract to call
+    @param _data Calldata
+    @dev Can only be called by operators
+    """
+    assert self.operators[msg.sender]
```

Comment:
curious why do we need a local state flag instead of calling the factory operators state? 

seems at first glance it could be out of sync?

Reply:
So vest recipients cant get rugged on their already received vests. Otherwise factory owner can add a malicious operator that transfers all their tokens back to them.

Reply:
understood